hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/commit/AbstractYarnClusterITest.java

```diff
@@ -61,45 +69,70 @@
  * If two subclasses of this class are instantiated in the same JVM, in order,
  * they are guaranteed to be isolated.
  *
- * History: this is a superclass extracted from
- * {@link AbstractITCommitMRJob} while adding support for testing terasorting.
- *
  */
 public abstract class AbstractYarnClusterITest extends AbstractCommitITest {
 
   private static final Logger LOG =
       LoggerFactory.getLogger(AbstractYarnClusterITest.class);
 
-  private static final int TEST_FILE_COUNT = 2;
-  private static final int SCALE_TEST_FILE_COUNT = 50;
+  private static final int TEST_FILE_COUNT = 1;
+  private static final int SCALE_TEST_FILE_COUNT = 2;
 
-  public static final int SCALE_TEST_KEYS = 1000;
+  public static final int SCALE_TEST_KEYS = 100;
   public static final int BASE_TEST_KEYS = 10;
 
   private boolean scaleTest;
 
-  private boolean uniqueFilenames = false;
+  /**
+   * The static cluster binding with the lifecycle of this test; served
+   * through instance-level methods for sharing across methods in the
+   * suite.
+   */
+  @SuppressWarnings("StaticNonFinalField")
+  private static ClusterBinding clusterBinding;
```

Comment:
I'm not sure how well this is going to work, given there's at least 2 tests which are inheriting from this class.

Either could end up creating the cluster, and then terminating the cluster in '@AfterClass' without knowing what the other test is doing - and I think this will lead to failures depending on the timing of executing ITestTerasortOnS3A and ITestS3ACommitterMRJob.

These tests likely need there own clusters, and teardown methods. Or maybe the parameter can be made non-static, so that individual test classes don't mess with each other. (They get their own clusters as a result). Afaik, the individual test classes will still be able to share clusters.

Reply:
The test suite is not creating and destroying clusters in indivudual methods

AbstractYarnClusterITest 

1. stores the binding in a static method
1. Instantiates in the first test case setup encountered; this is designed to pick up all those command line customisation properties you were missing out on.
1. And it is destroyed in a static afterclass method

All subclasses of this test get that behaviour.

Reply:
Junit `@AfterClass` javadocs

```
 The @AfterClass methods declared in superclasses will be run after those of the current
 class, unless they are shadowed in the current class.
 ```
 
 Nothing to worry about then; good to remember that ordering too. BeforeClass and Before are the opposite: superclasses run before the subclasses. What isn't documented is the ordering of before/after methods within the same class...I wouldn't rely on anything there.

Reply:
I think I understand the concern I raised a little better now.

Will the two derived tests ever run in parallel on the same VM?
- If so, there's a problem with the static cluster being in the base class - since the cluster will be shared and shutdown when either of the tests complete.

If parallel tests will only use separate VMs - we should be OK.

If a single VM is used to run the tests serially - the "clusterBinding=null" is an important change, which I think you've already made.

Reply:
now we have parameterized tests there will never be >1 CommitMRJob
test running in parallel. That was a problem we had before, where you could have multiple ITest*CommitMRJob test runs spawning so many processes that your dev box comes to a halt if you do a many threaded run. With these settings the delegation token test runs can overlap with the CommitMRJob sequence -i plan to follow up this patch with parameterization there too.

As all the tests have opted not to use a mini HDFS cluster we avoid the extra overhead of those processes too